src/value/borrowed.rs

```diff
@@ -137,6 +133,18 @@ impl<'val> SymbolToken for BorrowedSymbolToken<'val> {
     fn source(&self) -> Option<&Self::ImportSource> {
         self.source.as_ref()
     }
+
+    fn with_text(self, text: &'static str) -> Self {
+        self.with_text_borrowed(text)
+    }
+
+    fn with_local_sid(self, local_sid: SymbolId) -> Self {
+        self.with_local_sid_borrrowed(local_sid)
+    }
+
+    fn with_source(self, table: &'static str, sid: SymbolId) -> Self {
+        self.with_source_borrowed(table, sid)
+    }
```

Comment:
Let's make sure we have an issue to deal with this `'static` lifetime problem the APIs will be very marginal with this...